sdk/core/core-amqp/src/retry.ts

```diff
@@ -153,7 +153,7 @@ function calculateDelay(
     return Math.min(incrementDelta, maxRetryDelayInMs);
   }
 
-  return retryDelayInMs;
+  return Math.min(retryDelayInMs, maxRetryDelayInMs);
```

Comment:
I found that we didn't limit the delay in Fixed mode. I think we should respect in Fixed mode too.

Reply:
This got me thinking. Should we allow specifying a delay larger than the default max, if maxRetryDelayInMs is not specified?

Reply:
I got it now. It doesn't make sense for fixed mode because the delay is not changing.  Will remove